extensions/amp-date-display/1.0/amp-date-display.js

```diff
@@ -86,14 +87,33 @@ AmpDateDisplay['passthrough'] = true;
 
 /** @override */
 AmpDateDisplay['props'] = {
+  'datetime': {
+    attrs: ['datetime', 'timestamp-ms', 'timestamp-seconds', 'offset-seconds'],
+    parseAttrs: parseDateAttrs,
+  },
   'displayIn': {attr: 'display-in'},
-  'offsetSeconds': {attr: 'offset-seconds', type: 'number'},
   'locale': {attr: 'locale'},
-  'datetime': {attr: 'datetime'},
-  'timestampMs': {attr: 'timestamp-ms', type: 'number'},
-  'timestampSeconds': {attr: 'timestamp-seconds', type: 'number'},
 };
 
+/**
+ * @param {!Element} element
+ * @return {?number}
+ * @visibleForTesting
+ */
+export function parseDateAttrs(element) {
+  const datetimeStr = element.getAttribute('datetime');
+  const datetime = parseDate(datetimeStr);
+  const timestampMs = Number(element.getAttribute('timestamp-ms'));
```

Comment:
Are we ignoring the possibility of `NaN` because it will resolve itself in the chained `||` / assertion?

Reply:
Yes.